src/ref.js

```diff
@@ -0,0 +1,34 @@
+import isPlainObject from './helpers/isPlainObject';
+import wrapValidator from './helpers/wrapValidator';
+
+function isNewRef(prop) {
+  if (!isPlainObject(prop)) {
+    return false;
+  }
+  const ownProperties = Object.getOwnPropertyNames(prop);
+  return ownProperties.length === 1 && ownProperties[0] === 'current';
+}
+
+function ref(props, propName, componentName) {
+  const { [propName]: value } = props;
+
+  if (value == null || typeof value === 'function' || isNewRef(value)) {
+    return null;
+  }
+
+  return new Error(`${propName} in ${componentName} must be a ref`);
+}
+
+function requiredRef(props, propName, componentName) {
+  const { [propName]: value } = props;
+
+  if (value == null) {
+    return new Error(`${propName} in ${componentName} must be a ref`);
```

Comment:
nbd but typically i implement the error in the required one, and have the optional one call into it - rather than repeating the `new Error` part

Reply:
Ah, nice. I had it reversed, but doing it the way you suggested is nice